packages/roosterjs-content-model/lib/publicApi/utils/formatSegmentWithContentModel.ts

```diff
@@ -48,19 +58,13 @@ export function formatSegmentWithContentModel(
                 toggleStyleCallback(format, !isTurningOff, segment)
             );
 
-            const isCollapsedSelection =
-                segments.length == 1 && segments[0].segmentType == 'SelectionMarker';
-
             if (!pendingFormat && isCollapsedSelection) {
                 editor.setPendingFormat(segments[0].format);
-            }
-
-            if (isCollapsedSelection) {
```

Comment:
Can I get a specific function that should return? Pending state was causing a lot of formatting not to work.

Reply:
When we have changed something to the content model, return true. Otherwise return false.

For example:
1. when there is an expanded selection, we will change its format, so return true
2. when there is only collapsed selection, but we successfully found a word to format, so we will format the word, so still return true
3. when there is only collapsed selection, and no word is found, so we will update a pending format, but we didn't do any change to the content model itself, so return false.